xarray/backends/common.py

```diff
@@ -136,14 +136,15 @@ def _find_absolute_paths(
     def _normalize_path_list(
         lpaths: NestedSequence[str | os.PathLike],
     ) -> NestedSequence[str]:
-        return [
-            (
-                _normalize_path(p)
-                if isinstance(p, str | os.PathLike)
-                else _normalize_path_list(p)
-            )
-            for p in lpaths
-        ]
+        paths = []
+        for p in lpaths:
+            if isinstance(p, str | os.PathLike):
+                paths.append(_normalize_path(p))
+            elif isinstance(p, list):
+                paths.append(_normalize_path_list(p))  # type: ignore[arg-type]
+            else:
```

Comment:
Can you tell me what the expected type here is?
The type hints only "allow" str, os.PathLike or a list of these.

Reply:
Basically whatever 

```
fs = fsspec.filesystem("file")
fs.open(my_file)
```

returns

Reply:
FYI trying to determine what type `fs.open` returns could be a massive rabbit hole - see https://github.com/fsspec/filesystem_spec/issues/579#issuecomment-1140377495

Reply:
Ok, I guess a minimal protocol similar to how pandas handles it should be the solution then.